validation_framework/deployer/coe/kubernetes.py

```diff
@@ -0,0 +1,279 @@
+            version=version,
+            hostname=self.device.hostname
+        )
+
+        if self.include_peripherals:
+            for peripheral in self.peripherals:
+                cmd_arg = f' --set peripheralManager{peripheral}=true '
+                install_image_cmd = install_image_cmd + cmd_arg
+
+        envs_configuration: dict = self.engine_configuration.model_dump(by_alias=True)
+        if extra_envs:
+            envs_configuration.update(extra_envs)
+
+        self.logger.info(f'Starting NuvlaEdge with UUID: {self.nuvla_uuid} with'
+                         f' configuration: '
+                         f'\n\n {json.dumps(envs_configuration, indent=4)} \n')
+        self.device.run_sudo_command(install_image_cmd, envs=envs_configuration)
+
+        self.namespace = self.__get_current_nuvlaedge_namespace_running()
+
+        self.cred_check_thread = CertificateSignCheck(device_config=self.device.target_config,
+                                                      namespace=self.namespace)
+        self.cred_check_thread.start()
+
+        self.logger.info('Device start command executed')
+
+    def add_peripheral(self):
+        pass
+
+    def remove_peripheral(self):
+        pass
+
+    def stop_engine(self):
+        self.purge_engine()
+
+    def get_authorized_keys(self):
+        return self.device.run_sudo_command("sudo cat /root/.ssh/authorized_keys")
+
+    def get_engine_logs(self):
+        self.logger.info(f'Retrieving Log files from engine run with UUID: {self.engine_configuration.nuvlaedge_uuid}')
+        get_pods_cmd = (f'sudo kubectl get pods -n {self.namespace} -o json | '
+                        'jq \'.items[] | {name: .metadata.name}\'')
+        result: Result = self.device.run_sudo_command(get_pods_cmd)
+        if result.failed or result.stdout == '':
+            return
+
+        # 1. Get all pods in the concerned namespace
+        self.logger.debug(f'Extracting logs from: \n\n\t{result.stdout}\n')
+        pods = json.loads(result.stdout)
+
+        # 2. Iteratively copy logs from kubernetes pods
+        #    to /tmp/<new_folder> and chmod before transferring it to the running machine
+        self.device.run_sudo_command(
+            f'sudo mkdir -p /tmp/{self.engine_configuration.compose_project_name}')
+        local_tmp_path: Path = Path(f'/tmp/{self.engine_configuration.compose_project_name}')
+        local_tmp_path.mkdir(parents=True, exist_ok=True)
+
+        for pod in pods:
+            pod_name = pod.get("name")
+            get_logs_cmd = (f'sudo kubectl logs -n {self.namespace} {pod_name} >> '
+                            f'/tmp/{self.engine_configuration.compose_project_name}/{pod_name}.log')
+            self.device.run_sudo_command(get_logs_cmd)
+
+            self.device.download_remote_file(remote_file_path=f'/tmp/{pod_name}.log',
+                                             local_file_path=local_tmp_path / (pod_name + '.log'))
```

Comment:
This looks simpler than docker but I'd apply the same principle of simplicity on methods

Reply:
I hope I understand correctly, we should make the docker collection of logs similar to the Kubernetes one right?

Reply:
I meant just to follow the same principle of single purpose by separating the single container logs retrieval from the batch retrieval so it can be accessed in the future and better tested and it seems you did it already so fantastic :)